multiapps-controller-persistence/src/main/java/org/cloudfoundry/multiapps/controller/persistence/services/ObjectStoreFileStorage.java

```diff
@@ -101,18 +102,37 @@ public int deleteFilesModifiedBefore(LocalDateTime modificationTime) {
     public <T> T processFileContent(String space, String id, FileContentProcessor<T> fileContentProcessor) throws FileStorageException {
         FileEntry fileEntry = createFileEntry(space, id);
         try {
-            Blob blob = getBlobWithRetries(fileEntry, 3);
-            if (blob == null) {
-                throw new FileStorageException(MessageFormat.format(Messages.FILE_WITH_ID_AND_SPACE_DOES_NOT_EXIST, fileEntry.getId(),
-                                                                    fileEntry.getSpace()));
-            }
-            Payload payload = blob.getPayload();
+            Payload payload = getBlobPayload(fileEntry);
             return processContent(fileContentProcessor, payload);
         } catch (Exception e) {
             throw new FileStorageException(e);
         }
     }
 
+    private Payload getBlobPayload(FileEntry fileEntry) throws FileStorageException {
+        Blob blob = getBlobWithRetries(fileEntry, 3);
+        if (blob == null) {
+            throw new FileStorageException(MessageFormat.format(Messages.FILE_WITH_ID_AND_SPACE_DOES_NOT_EXIST, fileEntry.getId(),
+                                                                fileEntry.getSpace()));
+        }
+        return blob.getPayload();
+    }
+
+    @Override
+    public InputStream openInputStream(String space, String id) throws FileStorageException {
+        FileEntry fileEntry = createFileEntry(space, id);
+        Payload payload = getBlobPayload(fileEntry);
```

Comment:
Doesn't this load the entire payload in memory?

Reply:
No, in order to read it you have to open a stream